code/modules/paperwork/photocopier.dm

```diff
@@ -425,57 +427,41 @@
 	default_unfasten_wrench(user, tool)
 	return TOOL_ACT_TOOLTYPE_SUCCESS
 
-/obj/machinery/photocopier/attackby(obj/item/O, mob/user, params)
-	if(istype(O, /obj/item/paper))
-		if(copier_empty())
-			if(!user.temporarilyRemoveItemFromInventory(O))
-				return
-			paper_copy = O
-			do_insertion(O, user)
-		else
-			to_chat(user, span_warning("There is already something in [src]!"))
-
-	else if(istype(O, /obj/item/photo))
-		if(copier_empty())
-			if(!user.temporarilyRemoveItemFromInventory(O))
-				return
-			photo_copy = O
-			do_insertion(O, user)
-		else
-			to_chat(user, span_warning("There is already something in [src]!"))
+/obj/machinery/photocopier/attackby(obj/item/object, mob/user, params)
+	if(istype(object, /obj/item/paper))
+		insert_copy_object(object, user, PAPER_COPY_TYPE)
+	else if(istype(object, /obj/item/photo))
+		insert_copy_object(object, user, PHOTO_COPY_TYPE)
+	else if(istype(object, /obj/item/documents))
+		insert_copy_object(object, user, DOCUMENT_COPY_TYPE)
 
-	else if(istype(O, /obj/item/documents))
-		if(copier_empty())
-			if(!user.temporarilyRemoveItemFromInventory(O))
-				return
-			document_copy = O
-			do_insertion(O, user)
-		else
-			to_chat(user, span_warning("There is already something in [src]!"))
-
-	else if(istype(O, /obj/item/toner))
+	else if(istype(object, /obj/item/toner))
 		if(toner_cartridge)
 			to_chat(user, span_warning("[src] already has a toner cartridge inserted. Remove that one first."))
 			return
-		O.forceMove(src)
-		toner_cartridge = O
-		to_chat(user, span_notice("You insert [O] into [src]."))
+		object.forceMove(src)
+		toner_cartridge = object
+		to_chat(user, span_notice("You insert [object] into [src]."))
 
-	else if(istype(O, /obj/item/areaeditor/blueprints))
+	else if(istype(object, /obj/item/areaeditor/blueprints))
 		to_chat(user, span_warning("The Blueprint is too large to put into the copier. You need to find something else to record the document."))
-	else if(istype(O, /obj/item/paperwork))
-		if(copier_empty())
-			if(istype(O, /obj/item/paperwork/photocopy)) //No infinite paper chain. You need the original paperwork to make more copies.
-				to_chat(user, span_warning("The [O] is far too messy to produce a good copy!"))
-			else
-				if(!user.temporarilyRemoveItemFromInventory(O))
-					return
-				paperwork_copy = O
-				do_insertion(O, user)
+	else if(istype(object, /obj/item/paperwork))
+		if(istype(object, /obj/item/paperwork/photocopy)) //No infinite paper chain. You need the original paperwork to make more copies.
+			to_chat(user, span_warning("The [object] is far too messy to produce a good copy!"))
 		else
-			to_chat(user, span_warning("There is already something in [src]!"))
+			insert_copy_object(object, user, PAPERWORK_COPY_TYPE)
 		return ..()
```

Comment:
I don't know why this is only being used on paperwork, should I remove this or just escalate it so it occurs on all attackby's

Reply:
(the return ..())